tests/capture/input_generators/test_attack_point_iterator.py

```diff
@@ -145,3 +147,245 @@ def test_attack_point_iterator_unrestricted_generator_all_args_len():
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = list(build_attack_points_iterator(config))
+    assert output == []
+
+
+def test_attack_point_iterator_repeat_zero_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 0,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == 0
+
+
+def test_attack_point_iterator_repeat_one():
+    config = {
+        "operation": "repeat",
+        "repetitions": 1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = list(iter(build_attack_points_iterator(config)))
+    assert output == list(
+        iter(build_attack_points_iterator(config["configuration"])))
+
+
+def test_attack_point_iterator_repeat_one_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == len(config["configuration"]["values"])
+
+
+def test_attack_point_iterator_repeat_two():
+    config = {
+        "operation": "repeat",
+        "repetitions": 2,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = list(iter(build_attack_points_iterator(config)))
+    assert output == list(
+        iter(build_attack_points_iterator(
+            config["configuration"]))) * config["repetitions"]
+
+
+def test_attack_point_iterator_repeat_two_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 2,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == config["repetitions"] * len(
+        config["configuration"]["values"])
+
+
+def test_attack_point_iterator_repeat_three_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 3,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == config["repetitions"] * len(
+        config["configuration"]["values"])
+
+
+def test_attack_point_iterator_repeat_infinite():
+    config = {
+        "operation": "repeat",
+        "repetitions": -1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1]
+        }
+    }
+    count = 0
```

Comment:
Variable `count` is effectively unused in this test.
Consider rewriting as a for-loop
```python
for _ in range(5):
 for value1, value2 in zip(...):
  assert value1 == value2
```

Reply:
Forgot it was an infinite loop here....

Reply:
I don't think this works because the second for loop is an infinite loop